### PR TITLE
[PAY-1412] Reset player state on app load

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Hard to repro this bug but I think this should fix it. The theory is that after a crash, the player state from before the crash was persisted, which caused `getHasTrack` player selector to erroneously report `true`.

Would like to get this in for the QA and see if we get any repros of the bug.

### Dragons

Adding stuff to root screen!

### How Has This Been Tested?

Hard to prove this makes the bug go away, but I did confirm this action only runs once on startup and that normal app behavior is unaffected.

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/root-screen/RootScreen.tsx [hunk 1]
```diff
@@ -77,6 +83,14 @@ export const RootScreen = ({
...
   }, [dispatch, isLoaded, accountStatus])
 
+  // Reset the player when the app is loaded for the first time. Fixes an issue
+  // where after a crash, the player would persist the previous state. PAY-1412.
+  useEffect(() => {
+    if (isLoaded) {
+      dispatch(playerActions.reset({ shouldAutoplay: false }))
+    }
+  }, [isLoaded, dispatch])
+
```
Comment (piazzatron) [inline packages/mobile/src/screens/root-screen/RootScreen.tsx:93]: Could we just put this dispatch on L68 instead of having a new useEffect?

**Review by piazzatron — APPROVED** [review]
Nice find

File: packages/mobile/src/screens/root-screen/RootScreen.tsx (hunk 1 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/root-screen/RootScreen.tsx:93]: I think we want to keep them separate bc we don't want to clear this state if a user was not signed in, then signs in? L68 runs when the user has successfully signed in only. We want this to run when the app loads for the first time regardless of sign-in state.

Comment (dharit-tan) [reply packages/mobile/src/screens/root-screen/RootScreen.tsx:93]: I guess could add `if (isLoaded)`-only check to the above useEffect. I could go either way lmk.

Comment (dharit-tan) [reply packages/mobile/src/screens/root-screen/RootScreen.tsx]: chatted with @piazzatron think he's right!

**Pull request merged**